frontend/packages/ux-editor/src/utils/component.ts

```diff
@@ -28,3 +30,43 @@ export enum AddressKeys {
   careOf = 'careOf',
   houseNumber = 'houseNumber',
 }
+
+export const changeTextResourceBinding = (
+  component: IFormComponent,
+  bindingKey: string,
+  resourceKey: string
+): IFormComponent => ({
+  ...component,
+  textResourceBindings: {
+    ...component.textResourceBindings,
+    [bindingKey]: resourceKey,
+  }
+});
+
+export const changeTitleBinding = (component: IFormComponent, resourceKey: string): IFormComponent =>
+  changeTextResourceBinding(component, 'title', resourceKey);
+
+export const changeDescriptionBinding = (component: IFormComponent, resourceKey: string): IFormComponent =>
+  changeTextResourceBinding(component, 'description', resourceKey);
+
+export const addOptionToComponent = <T extends IFormCheckboxComponent | IFormRadioButtonComponent>(
+  component: T,
+  option: IOptions
+): T => ({
+  ...component,
+  options: [
+    ...component.options,
+    option,
+  ],
+});
+
+export const changeComponentOptionLabel = <T extends IFormCheckboxComponent | IFormRadioButtonComponent>(
```

Comment:
Maybe it could be more readable?
OptionsComponent (instead of a generic type?)
Maybe IFormCheckboxComponent and IFormRadioButtonComponent can extend an OptionComponent that has the options prop?

Reply:
Good point. I see now that there is already a `IFormGenericOptionsComponent`, it's just not used for checkboxes and radio buttons.